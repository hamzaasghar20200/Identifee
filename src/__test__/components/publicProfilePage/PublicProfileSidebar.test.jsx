import { render } from '@testing-library/react';

import PublicProfileSidebar from '../../../components/publicProfilePage/PublicProfileSidebar';

describe('PublicProfileSidebar', () => {
  test('snapshot', () => {
    const { container } = render(<PublicProfileSidebar />);

    expect(container).toMatchSnapshot();
  });

  test('Should have an icon, an avatar', () => {
    const { container } = render(<PublicProfileSidebar />);

    const icon = container.getElementsByClassName(
      'material-icons-outlined text-white'
    )[0];

    expect(icon).toBeTruthy();

    const avatar = container.getElementsByClassName(
      'avatar avatar-xl avatar-circle avatar-border-dashed mr-3'
    )[0];

    expect(avatar).toBeTruthy();
  });

  test('Should have a text with userInfo name and a title', () => {
    const userInfo = {
      first_name: 'John',
      last_name: 'Doe',
      title: 'Manager',
    };

    const { container } = render(<PublicProfileSidebar userInfo={userInfo} />);

    const name = container.querySelector('h3[class="mb-2 text-white mt-3"]');

    expect(name.textContent).toBe(' John Doe ');

    const title = container.querySelector(
      'span[class="d-block text-gray-200 font-size-sm"]'
    );

    expect(title.textContent).toBe('Manager');
  });

  test('Should have a button with Schedule Call label and icon', () => {
    const { container } = render(<PublicProfileSidebar />);

    const icon = container.querySelectorAll(
      'button[class="btn btn-outline-light font-weight-medium mt-5"] > span:nth-child(1)'
    )[0];

    expect(icon.textContent).toBe('call');

    const button = container.querySelectorAll(
      'button[class="btn btn-outline-light font-weight-medium mt-5"] > span:nth-child(2)'
    )[0];

    expect(button.textContent).toBe('Schedule Call');
  });

  test('Should have a button with Your Bank Team label and icon', () => {
    const { container } = render(<PublicProfileSidebar />);

    const icon = container.querySelectorAll(
      'button[class="btn btn-outline-light font-weight-medium mt-5"] > span:nth-child(1)'
    )[1];

    expect(icon.textContent).toBe('group');

    const button = container.querySelectorAll(
      'button[class="btn btn-outline-light font-weight-medium mt-5"] > span:nth-child(2)'
    )[1];

    expect(button.textContent).toBe('Your Bank Team');
  });
});
