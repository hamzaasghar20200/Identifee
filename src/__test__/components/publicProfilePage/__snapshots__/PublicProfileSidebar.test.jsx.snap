// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicProfileSidebar snapshot 1`] = `
<div>
  <div
    class="col-md-3 col-lg-2 col-fixed d-flex flex-column align-items-center text-center px-4 py-6 col"
    style="background: rgb(12, 24, 61); min-height: 100vh;"
  >
    <span
      class="material-icons-outlined text-white"
      style="font-size: 74px;"
    >
      account_balance
    </span>
    <div
      class="mt-4 mt-md-10"
    >
      <div
        class="avatar avatar-xl avatar-circle avatar-border-dashed mr-3"
      >
        <div
          class="avatar avatar-sm avatar-circle avatar-soft-primary avatar-xl"
        >
          <span
            class="avatar-initials"
          >
            <span
              class="material-icons-outlined avatar-icon-font-size"
            >
              person
            </span>
          </span>
        </div>
      </div>
      <h3
        class="mb-2 text-white mt-3"
      >
         
        undefined undefined
         
      </h3>
      <span
        class="d-block text-gray-200 font-size-sm"
      />
      <button
        class="btn btn-outline-light font-weight-medium mt-5"
      >
        <span
          class="material-icons-outlined mr-1"
        >
          call
        </span>
        <span>
          Schedule Call
        </span>
      </button>
      <section
        class="alert-position mr-5"
      />
      <button
        class="btn btn-outline-light font-weight-medium mt-5"
      >
        <span
          class="material-icons-outlined mr-1"
        >
          group
        </span>
        <span>
          Your Bank Team
        </span>
      </button>
      <section
        class="alert-position mr-5"
      />
      <div
        class="mt-5 text-gray-400 font-size-sm"
      >
        <p
          class="font-size-md font-weight-medium mb-2"
        >
          For more information: 
        </p>
        <p
          class="mb-1"
        >
           
          undefined undefined
           
        </p>
        <p
          class="mb-1"
        />
        <p
          class="mb-0 text-lowercase"
        />
      </div>
    </div>
  </div>
</div>
`;
